File: UMA.App.IdentityManager/Role/Queries/CreateRoleQuery.cs
using MediatR;
using UMA.App.Common.AutoMapper;
using UMA.App.IdentityManager.Role.Models;

namespace UMA.App.IdentityManager.Role.Queries
{
    public class CreateRoleQuery : IRequest<CreateRoleResult>, IHaveCustomMapping
    {
        public string Name { get; set; }
        public string Description { get; set; }

        public void CreateMappings(AutoMapper.Profile configuration)
        {
            configuration.CreateMap<CreateRoleQuery, Domain.Identity.Role>();
        }
    }
}

File: UMA.Web.Angular/UMA-Web-Angular/src/app/_entities/role/query/createRole.query.ts
import { IRequest } from 'src/app/_models/interfaces/IRequest';
import { CreateRoleViewModel } from '../reponse/createRole.viewmodel';

export class CreateRoleQuery implements IRequest<CreateRoleViewModel> {
  _response?: CreateRoleViewModel;
  name: string;
  description: string;
}
